protostar/starknet_gateway/gateway_facade.py

```diff
@@ -281,3 +284,44 @@ def __init__(self, function_name: str):
 class ContractNotFoundException(ProtostarException):
     def __init__(self, contract_address: AddressRepresentation):
         super().__init__(f"Tried to call unknown contract:\n{contract_address}")
+
+
+class InvalidInputException(ProtostarException):
+    pass
+
+
+def validate_deploy_input(compiled_contract_json: str, inputs: List[int]) -> None:
+    def constructor_filter(element: Dict):
+        return element["type"] == "constructor"
+
+    def get_type_size(abi: List[Dict], typename: str):
+        if typename == "felt":
+            return 1
+
+        def type_filter(element: Dict):
+            return element["name"] == typename
+
+        [type_definition] = filter(type_filter, abi)
+        return type_definition["size"]
+
+    abi = json.loads(compiled_contract_json)["abi"]
+    [constructor] = filter(constructor_filter, abi)
+    expected_inputs = constructor["inputs"]
+
+    i = 0
+    for expected_input in expected_inputs:
+        raw_type_size = get_type_size(abi, expected_input["type"].rstrip("*"))
+        if expected_input["type"].endswith("*"):
+            size = inputs[i - 1] * raw_type_size
+        else:
+            size = raw_type_size
```

Comment:
Could this code use `parse_type` function from starkware/cairo/lang/compiler/parser.py ?

Reply:
I don't think it gives us what we need

Reply:
it will give you information if you have raw `felt` or a pointer to something. this will definitely be more robust and future proof (what happens if starkware changes pointer syntax?) then manual parsing